### ISSUE TITLE
Upgrade: 0.29.3 > 0.30.0 - NumberFormatException when starting metabase

### ISSUE THREAD

**Issue by jlindenbaum** [metabase/metabase#8297]
Environment:
- Ubuntu 16.04 LTS.
- CloudSQL running MySQL 5.6.x
- Java
```
openjdk version "9-internal"
OpenJDK Runtime Environment (build 9-internal+0-2016-04-14-195246.buildd.src)
OpenJDK 64-Bit Server VM (build 9-internal+0-2016-04-14-195246.buildd.src, mixed mode)
```

Steps to reproduce:
- Stop metabase 0.29.3
- Replace jar with 0.30.0
- Start metabase

Expected
- Migrations run
- Metabase starts

Actual
Exception message appears, metabase does not start.
```
08-13 09:46:29 INFO metabase.util :: Loading Metabase...
08-13 09:46:36 INFO util.encryption :: Saved credentials encryption is DISABLED for this Metabase instance. 🔓 
For more information, see https://www.metabase.com/docs/latest/operations-guide/start.html#encrypting-your-database-connection-details-at-rest
08-13 09:46:51 INFO metabase.core :: Starting Metabase in STANDALONE mode
08-13 09:46:51 INFO metabase.core :: Launching Embedded Jetty Webserver with config: 
 {:port 3000}

08-13 09:46:52 INFO metabase.core :: Starting Metabase version v0.30.0 (04836dc release-0.30.0) ...
08-13 09:46:52 INFO metabase.core :: System timezone is 'Etc/UTC' ...
08-13 09:46:52 ERROR metabase.core :: Metabase Initialization FAILED
java.lang.NumberFormatException: For input string: "9-internal"
	at java.lang.NumberFormatException.forInputString(java.base@9-internal/NumberFormatException.java:65)
	at java.lang.Integer.parseInt(java.base@9-internal/Integer.java:695)
	at java.lang.Integer.parseInt(java.base@9-internal/Integer.java:813)
	at metabase.util$is_java_9_or_higher_QMARK_.invokeStatic(util.clj:553)
	at metabase.util$is_java_9_or_higher_QMARK_.invoke(util.clj:550)
	at metabase.plugins$load_plugins_BANG_.invokeStatic(plugins.clj:77)
	at metabase.plugins$load_plugins_BANG_.invoke(plugins.clj:73)
	at metabase.core$init_BANG_.invokeStatic(core.clj:150)
	at metabase.core$init_BANG_.invoke(core.clj:138)
	at metabase.core$start_normally.invokeStatic(core.clj:245)
	at metabase.core$start_normally.invoke(core.clj:239)
	at metabase.core$_main.invokeStatic(core.clj:265)
	at metabase.core$_main.doInvoke(core.clj:260)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at metabase.core.main(Unknown Source)
08-13 09:46:52 INFO metabase.core :: Metabase Shutting Down ...
08-13 09:46:52 INFO metabase.core :: Metabase Shutdown COMPLETE
```

**Comment by jornh** [comment]
Thanks for the good error description! 🙂 

Pretty sure this is a duplicate of #8282

There are two possible workarounds for now: 
- Switch to a Java version with a decimal point like e.g `1.8.0_171` or `10.0.2` (JDK 9 is actually unsupported by Oracle now)
- Modify and build the Metabase source with adding two lines to also split on `-`
  So that https://github.com/metabase/metabase/blob/v0.30.0/src/metabase/util.clj#L553-L556
  becomes: 
```
  (when-let [java-major-version (some-> (System/getProperty "java.version")
                                        (s/split #"-")
                                        first
                                        (s/split #"\.")
                                        first
                                        Integer/parseInt)]

```

... or wait for a fix landing in an upcoming v0.30.1 ...

**Comment by salsakran** [comment]
dupe of #8282 and fixed in 0.31, as @jornh so helpfully mentioned =)

**Comment by jornh** [comment]
According to https://github.com/metabase/metabase/issues/8282#issuecomment-412640626 the issue was fixed in #8305 which we'll ship as part of our 0.30.1 bugfix release in the next few days.

@jlindenbaum I think you can close this issue.
